tests/backup/backup_upgrade_test.go

```diff
@@ -1546,3 +1546,347 @@ var _ = Describe("{PXBackupClusterUpgradeTest}", Label(TestCaseLabelsMap[PXBacku
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+
+			err = AddAzureApplicationClusters(BackupOrgID, cloudCredName, cloudCredUID, ctx)
+			dash.VerifyFatal(err, nil, "Creating source and destination cluster")
+
+			clusterStatus, err := Inst().Backup.GetClusterStatus(BackupOrgID, SourceClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", SourceClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", SourceClusterName))
+
+			sourceClusterUid, err = Inst().Backup.GetClusterUID(ctx, BackupOrgID, SourceClusterName)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Fetching [%s] cluster uid", SourceClusterName))
+
+			clusterStatus, err = Inst().Backup.GetClusterStatus(BackupOrgID, DestinationClusterName, ctx)
+			log.FailOnError(err, fmt.Sprintf("Fetching [%s] cluster status", DestinationClusterName))
+			dash.VerifyFatal(clusterStatus, api.ClusterInfo_StatusInfo_Online, fmt.Sprintf("Verifying if [%s] cluster is online", DestinationClusterName))
+		})
+
+		Step("Taking backup of application from source cluster before upgrade", func() {
+			log.InfoD("Taking backup of application from source cluster before upgrade")
+			ctx, err := backup.GetAdminCtxFromSecret()
+			log.FailOnError(err, "Fetching px-central-admin ctx")
+			if preUpgradePxBackupVersion.GreaterThanOrEqual(pxbackupVersion270) {
+				err = os.Setenv("BACKUP_TYPE", "native_csi")
+			} else {
+				err = os.Setenv("BACKUP_TYPE", "azure")
+			}
+			log.FailOnError(err, "Setting BACKUP_TYPE env variable")
+
+			preUpgradeBackupName = fmt.Sprintf("%s-%s", "autogenerated-backup-pre-upgrade", RandomString(4))
+			log.InfoD("creating backup [%s] in source cluster [%s] (%s), organization [%s], in backup location [%s]", preUpgradeBackupName, SourceClusterName, sourceClusterUid, BackupOrgID, backupLocationName)
+			err = CreateBackupWithValidation(ctx, preUpgradeBackupName, SourceClusterName, backupLocationName, backupLocationUID, scheduledAppContexts, labelSelectors, BackupOrgID, sourceClusterUid, "", "", "", "")
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Creation and Validation of backup [%s]", preUpgradeBackupName))
+		})
+
+		Step("Upgrade Px Backup", func() {
+			log.InfoD("Upgrade Px Backup to version %s", LatestPxBackupVersion)
+			err := PxBackupUpgrade(LatestPxBackupVersion)
+			dash.VerifyFatal(err, nil, "Verifying Px Backup upgrade completion")
+		})
+
+		Step("Upgrade the stork version", func() {
+			log.InfoD("Upgrade the stork version")
+			upgradeStorkImageStr = LatestStorkImage
+			log.Infof("Upgrading stork version on source cluster to %s ", upgradeStorkImageStr)
+			err := UpgradeStorkVersion(upgradeStorkImageStr)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Verification of stork version upgrade to - %s on source cluster", upgradeStorkImageStr))
+			//err = SetDestinationKubeConfig()
+			//log.FailOnError(err, "Switching context to destination cluster failed")
```

Comment:
If not needed can we remove if needed then please add a TODO comment/

Reply:
This was commented out just for testing purpose because I just had one source cluster

Reply:
The comment will be removed before merging